src/stream/src/executor/managed_state/aggregation/mod.rs

```diff
@@ -91,21 +92,28 @@ impl<S: StateStore> ManagedStateImpl<S> {
     }
 
     /// Flush the internal state to a write batch.
-    pub fn flush(&mut self, write_batch: &mut WriteBatch<S>) -> Result<()> {
+    pub async fn flush(
+        &mut self,
+        write_batch: &mut WriteBatch<S>,
+        state_table: &mut StateTable<S>,
+    ) -> Result<()> {
         match self {
-            Self::Value(state) => state.flush(write_batch),
+            Self::Value(state) => state.flush(write_batch, state_table).await,
```

Comment:
Yes it's the same as what we discussed above?

So if we store the ownership of state_table in it, we can hide state_table from executors. But we need a `Rc<RefCell<>>` to wrap it.

Previous RFC proposed by @st1page : https://singularity-data.quip.com/buN6ASPdobmk/RFC-Split-Dirty-State-and-Cache needs to seperate cache and states. And in here, the cache is the AggState, the state is the `state_table`.  If we store `state_table` in AggState, we can not control them seperately?

Reply:
But the doc is mainly focused on HashJoin or Max/Min of HashAgg. For value states, we could consider to store state_table inside it. But we have to wrap a `Rc<RefCell<>>`. I think it's acceptable.

Reply:
After a second thought, I think store state table inside each state is not good enough. 

If we store state table inside each state, then we must call `.commit` of state_table in each `.flush` call. This can not batch commit -- given that one state table might have mutiple states shared, we should batch commit these states. This also inconsistent with previous WriteBatch flush mechanism.

If we not store state table, we can batch commit the state table in executor, just like this PR.